src/api/v1/users/volunteers/__tests__/volunteer_logs.test.integration.ts

```diff
@@ -0,0 +1,361 @@
+import * as Hapi from 'hapi';
+import * as Knex from 'knex';
+import * as moment from 'moment';
+import { omit } from 'ramda';
+import { init } from '../../../../../server';
+import { getConfig } from '../../../../../../config';
+import { getTrx } from '../../../../../../tests/utils/database';
+import { User, Users, Organisation, Organisations, VolunteerLog } from '../../../../../models';
+
+
+describe('API /users/me/volunteer-logs', () => {
+  let server: Hapi.Server;
+  let knex: Knex;
+  let trx: Knex.Transaction;
+  let user: User;
+  let organisation: Organisation;
+  const config = getConfig(process.env.NODE_ENV);
+
+  beforeAll(async () => {
+    server = await init(config);
+    knex = server.app.knex;
+
+    user = await Users.getOne(knex, { where: { name: 'Emma Emmerich' } });
+    organisation = await Organisations.getOne(knex, { where: { name: 'Black Mesa Research' } });
+  });
+
+  afterAll(async () => {
+    await server.shutdown(true);
+  });
+
+  beforeEach(async () => {
+    trx = await getTrx(knex);
+    server.app.knex = trx;
+  });
+
+  afterEach(async () => {
+    await trx.rollback();
+    server.app.knex = knex;
+  });
+
+  describe('GET /users/me/volunteers-logs', () => {
+    test('can get own logs', async () => {
+      const res = await server.inject({
+        method: 'GET',
+        url: '/v1/users/volunteers/me/volunteer-logs',
+        credentials: {
+          user,
+          organisation,
+          scope: ['volunteer_logs-parent:read'],
+        },
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect((<any> res.result).result).toHaveLength(7);
+      expect(res.result).toEqual({
+        result: expect.arrayContaining([
+          expect.objectContaining({
+            organisationId: organisation.id,
+            userId: user.id,
+            duration: { minutes: 30 },
+          }),
+        ]),
+      });
+    });
+
+    test('can get own logs filtered by date', async () => {
+      const since = moment().day(-6).toDate().toISOString();
+      const until = moment().day(-5).toDate().toISOString();
+
+      const res = await server.inject({
+        method: 'GET',
+        url: `/v1/users/volunteers/me/volunteer-logs?since=${since}&until=${until}`,
+        credentials: {
+          user,
+          organisation,
+          scope: ['volunteer_logs-parent:read'],
+        },
+      });
+
+      expect(res.statusCode).toBe(200);
+      expect((<any> res.result).result).toHaveLength(1);
+      expect(res.result).toEqual({
+        result: expect.arrayContaining([
+          expect.objectContaining({
+            organisationId: organisation.id,
+            userId: user.id,
+            activity: 'Outdoor and practical work',
+            duration: { hours: 5 },
+          }),
+        ]),
+      });
+    });
+
+    test('can get no logs for non-volunteer', async () => {
```

Comment:
So, under our new error handling plan...

Would the method in this handler throw an error, and this route return a 404?

Reply:
Yeah this was unchanged from before the URL change. When it was `/users/me/volunteer-logs`, a `200 []` response made sense because `/users/me` is an existing resource, but one that has no `volunteer-logs` associated to it.

Now that the URL has changed, `/users/volunteers` is a collection that does not include `me` if that `me` is not a volunteer, so `/users/volunteers/me` (and all sub-routes) should be a 404.

As far as how this 404 is generated, we could either put a check in the model method and/or we could check the value of `role` in the auth credentials.